### PR TITLE
[Logs] Update docs processing docs

### PR DESCRIPTION
**Pull request by ruflin**
This enhances the current logs ingest processing doc.

Initially opening as draft to have some more conversations on this first.

### CODE REVIEW

File: docs/en/observability/logs-stream.asciidoc
Comment (ruflin) [inline docs/en/observability/logs-stream.asciidoc:345]: @eyalkoren @felixbarny I suggest we recommend users to add the ecs template by default?

File: docs/en/observability/logs-stream.asciidoc
Comment (mdbirnstiehl) [inline docs/en/observability/logs-stream.asciidoc:256]:
```suggestion
To extract the `@timestamp` field from the example log, use an ingest pipeline with a dissect processor. Ingest pipelines in {es} are used to process incoming documents. The {ref}/dissect-processor.html[dissect processor] is one of the available processors that extracts structured fields from your unstructured log message based the pattern you set. In the following example command, the dissect processor extracts the timestamp to the `@timestamp` field.
```

File: docs/en/observability/logs-stream.asciidoc
Comment (ruflin) [reply docs/en/observability/logs-stream.asciidoc:168]: yes

File: docs/en/observability/logs-stream.asciidoc
Comment (ruflin) [reply docs/en/observability/logs-stream.asciidoc:342]: Makes sense as I switched to a pattern.

File: docs/en/observability/logs-stream.asciidoc
Comment (ruflin) [reply docs/en/observability/logs-stream.asciidoc:345]: I'll add it to the template. It means we also have an opportunity here to quickly explain why these are all here.

File: docs/en/observability/logs-stream.asciidoc
Comment (ruflin) [inline docs/en/observability/logs-stream.asciidoc:350]: @mdbirnstiehl In a follow up PR, we should explain in details what these are doing. I would like to quickly sync with you on this to discuss how we describe this best. A quick summary to get it storated:

**Comment by ruflin** [comment]
I marked this a ready for review. I think we can get this in in the current state (after review) and follow up on the open conversation around the exact templates and what they do.

File: docs/en/observability/logs-stream.asciidoc
Comment (mdbirnstiehl) [inline docs/en/observability/logs-stream.asciidoc:378]:
```suggestion
- `logs-settings`* – Sets the general settings for logs data streams including the default lifecycle policy and default pipeline: 
-- The default lifecycle policy rolls over when the primary shard reaches 50 GB or after 30 days.
-- The default pipeline:
--- Sets a `@timestamp` if there isn't one using the ingest timestamp.
--- Places a hook for the `logs@custom` pipeline. If a `logs@custom` pipeline is installed, it's applied to logs ingested into this data stream.
```

File: docs/en/observability/logs-stream.asciidoc
Comment (mdbirnstiehl) [inline docs/en/observability/logs-stream.asciidoc:379]:
```suggestion
-- Sets the {ref}/ignore-malformed.html[`ignore_malformed`] flag to `true`. If a field in the log document contains an incorrect value type and the field's mapping type supports this flag, the document is still processed.
```

File: docs/en/observability/logs-stream.asciidoc
Comment (mdbirnstiehl) [inline docs/en/observability/logs-stream.asciidoc:380]:
```suggestion
- `logs@custom` – a predefined component template that is not installed by default. Use this name to install a custom component template if you wish to override or extend any of the default mappings or settings.
```

File: docs/en/observability/logs-stream.asciidoc
Comment (mdbirnstiehl) [inline docs/en/observability/logs-stream.asciidoc:381]:
```suggestion
* *`ecs@dynamic_templates`* - dynamic templates that ensure automatic compliance of your data stream mappings with the {ecs-ref}/ecs-reference.html[Elastic Common Schema (ECS)]
```
```suggestion
* `ecs@dynamic_templates` – dynamic templates that automatically ensure you data stream mappings comply with the {ecs-ref}/ecs-reference.html[Elastic Common Schema (ECS)].
```

**Review by mdbirnstiehl — APPROVED** [review]
Did a quick review of the updates, but other than that, LGTM. I also added the 8.9 backport, assuming we want these updates in the current docs as well as 8.10.

File: docs/en/observability/logs-stream.asciidoc
Comment (ruflin) [reply docs/en/observability/logs-stream.asciidoc:350]: Thanks @eyalkoren . Not sure what you mean by dynamic vs perma links?

**Comment by ruflin** [comment]
@mdbirnstiehl Could you push this over the line by getting the merge conflicts resolved and merge it? Not getting to it at the moment but would like to see this in and then keep iterating.

File: docs/en/observability/logs-stream.asciidoc
Comment (eyalkoren) [reply docs/en/observability/logs-stream.asciidoc:350]: A link to the corresponding template file in main (e.g. [`logs-mappings.json`](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/template-resources/src/main/resources/logs-mappings.json)) would be useful, but may break docs build if changed/moved (just like [happened recently](https://github.com/elastic/elasticsearch/pull/97763)).
A link to a permanent URL (e.g. [`logs-mappings.json`](https://github.com/elastic/elasticsearch/blob/1955bd8ad421e4acc37dbb4e948c5ff80fce4131/x-pack/plugin/core/template-resources/src/main/resources/logs-mappings.json)) will not be very useful.

Comment (ruflin) [reply docs/en/observability/logs-stream.asciidoc:345]: I really like the idea of being able to link to it or copy it over and update it from time to time. Having the templates in the docs directly will also allow to explain these better. Maybe we need a page where these are explained, a dedicate page? If we have this, how would we keep it up-to-date there? Seems like one of these common issues on how to keep docs and code in sync.

Comment (eyalkoren) [reply docs/en/observability/logs-stream.asciidoc:345]: If we come up with a method to keep docs up-to-date somehow, maybe we can use permanent links and keep them up to date.

Comment (ruflin) [reply docs/en/observability/logs-stream.asciidoc:345]: @mdbirnstiehl @bmorelli25 Has something like this been done in the past?

Comment (mdbirnstiehl) [reply docs/en/observability/logs-stream.asciidoc:345]: @bmorelli25 can answer maybe better than me if this fits the situation. BI think in the APM realm some files that have ties to docs require a review from obs-docs before merging to prevent the file from being updated without the corresponding docs. Not sure if we could do something like that in this instance.

**Pull request merged**
